src/librustdoc/html/markdown.rs

```diff
@@ -560,7 +560,8 @@ impl<'a, 'b, 'ids, I: Iterator<Item = SpannedEvent<'a>>> Iterator
                 self.buf.push_front((Event::Html(format!("{} ", sec).into()), 0..0));
             }
 
-            let level = std::cmp::min(level + (self.heading_offset as u32), MAX_HEADER_LEVEL);
+            let level =
+                std::cmp::min(level as u32 + (self.heading_offset as u32), MAX_HEADER_LEVEL);
```

Comment:
Is level a `usize`? If so, please remove the new `as` and replace the other one with a cast to usize.

Reply:
No, it's a `HeadingLevel` (which is why I need the cast).

Reply:
I know, I was suggesting something like this:

```suggestion
                std::cmp::min(level + (self.heading_offset as usize), MAX_HEADER_LEVEL);
```

Reply:
Silly me, you're absolutely right.

Reply:
Hum actually no, just like we need to convert `self.heading_offset`, we need to convert `level` as well...

Reply:
Eh, it's fine. I want to change `HeadingLevel` at some point anyway.